static/js/millennium_diff_table.js

```diff
@@ -235,38 +235,29 @@ function matchResources(conformanceData, capabilityStatementData) {
 }
 
 /**
- * Merges the two sorted arrays alphabetically.
+ * Merges the two sorted arrays alphabetically. Entries are ordered by their R4 name first, if no R4 resource
+ * exists then the DSTU2 is used.
  * @param {array} includesR4 - The array of resource objects that contain R4 resources.
  * @param {array} dstu2Only - The array of resource objects that only contain DSTU2 resources.
  * @returns {array} The resultant array after merging.
  */
 function mergeSortedArrays(includesR4, dstu2Only) {
   let result = [];
-  let i = 0;
-  let j = 0;
-
-  while (i < includesR4.length && j < dstu2Only.length) {
-    if (includesR4[i].hasOwnProperty('r4Resources') && !includesR4[i].hasOwnProperty('dstu2Resources')) {
-      if (includesR4[i].r4Resources[0].resourceName < dstu2Only[j].dstu2Resources[0].resourceName) {
-        result.push(includesR4[i]);
-        i += 1;
-      }
-      else {
-        result.push(dstu2Only[j]);
-        j += 1;
-      }
-    }
-    else {
-      if (includesR4[i].dstu2Resources[0].resourceName < dstu2Only[j].dstu2Resources[0].resourceName) {
-        result.push(includesR4[i]);
-        i += 1;
-      }
-      else {
-        result.push(dstu2Only[j]);
-        j += 1;
+
+  for(let x = 0; x < includesR4.length; x += 1) {
+    let r4Record = includesR4[x];
+
+    for (let y = 0; y < dstu2Only.length; y += 1) {
```

Comment:
nit: You'll speed up the search a bit if you define variables for `includesR4.length` and `dstu2Only.length` before starting the loop as well. The current syntax is going to check the length each time you go through the loop.

Reply:
dstu2Only.length needs to stay as is since we are splicing the array and removing an element

Reply:
Iterating on a changing array seems like a bad idea. It would be better to use another variable as a starting point for "y", but this is fine.

Reply:
https://github.com/cerner/fhir.cerner.com/commit/e52ecc574fd6f521216755e0410a2911dd8ddb5a